packages/core/lib/src/widgets/ui/app_gap.dart

```diff
@@ -0,0 +1,67 @@
+import 'package:core/src/styles/app_spacing.dart';
+import 'package:flutter/material.dart';
+
+/// A wrapper around [SizedBox] to create a Gap widget
+/// that can be used to separate other widgets
+///
+/// Comes with use cases for different sizes
+/// as horizontal `h` or vertical `v` spacing
+///
+/// Usage Examples:
+/// Instead of `const SizedBox(width: AppSpacing.sm)`
+/// We can use `const AppGap.hSmall()`
+///
+/// Instead of `const SizedBox(height: AppSpacing.md)`
+/// We can use `const AppGap.v()`
```

Comment:
@jenshor would love to get your opinion on this before I continue with it and make the changes across the app.

The idea is to create a simplified implementation of the `Gap` concept using the `AppSpacing` class and without the package.
I'm not very fond of the naming of the constructors and would love to hear your suggestions. We can also leave the naming as is now to focus on other things if you're okay with this implementation.

Reply:
Haha, yeah the constructors look pretty ugly. 
I understand that it's not necessary to set the width if you want to do a horizontal gap but is there any harm to set the width to the same as the height? 
I always use a "square" SizedBox and I have never had any problems with it. 

instead of using `v` and `h` we could use `horizontal` and `vertical` instead which would make it more readable and instead of using `xs` which currently translates to `xSmall` we could use `extraSmall` instead. 

But in the end it doesn't really matter too much.

Reply:
I agree we can use a square `SizedBox` I also can't think of a reason that it would be a problem.

So this way we don't have to use `horizontal` or `vertical`.
I can change the `xSmall` to `extraSmall` but what if it's `xxs` do you think it's okay to name it `extraExtraSmall` 😅?

Reply:
Or I think we can do something similar to the `Avatar` widget and have names like `small`, `smaller` and `smallest`. What do you think?

Reply:
We can also use `xs` and `xxs` as long as we don't get sth that looks like `vxs` 😀

I think `small` `smaller` `smallest` is a bit weird since that sounds like `small` `medium` `large` but in the end we could rename this and the sizing is clear in both examples.. so it doesn't really matter 🤷🏼‍♂️